pilot/pkg/networking/core/v1alpha3/networkfilter.go

```diff
@@ -136,6 +138,9 @@ func buildOutboundNetworkFiltersWithWeightedClusters(node *model.Proxy, routes [
 
 	// For weighted clusters set hash policy if any of the upstream destinations have sourceIP.
 	maybeSetHashPolicy(destinationRule, tcpProxy, "")
+	// In case of weighted clusters, tunneling config for a subset is ignored,
+	// because it is set on listener, not on a cluster.
+	tunnelingconfig.Apply(tcpProxy, destinationRule, "")
```

Comment:
this is pretty unexpected since we are basically saying ij a weighted cluster, for some DR fields the first cluster applies to all of them. However, this behavior was introduced in #35014 and is only extended here...